modules/secrets-store-azure.adoc

```diff
@@ -0,0 +1,112 @@
+// Module included in the following assemblies:
+//
+// * nodes/pods/nodes-pods-secrets-store.adoc
+
+:_content-type: PROCEDURE
+[id="secrets-store-azure_{context}"]
+= Mounting secrets from an external secrets store to a CSI volume
+
+TODO: intro
+
+This procedure assumes that you are using Azure Key Vault as the secrets store provider.
+
+.Prerequisites
+
+* You have installed the {secrets-store-operator}. See _Installing the Secrets Store CSI Driver_ for instructions.
+* You have installed a secrets store provider. See the appropriate vendor documentation for instructions. This procedure assumes that you have installed the link:https://azure.github.io/secrets-store-csi-driver-provider-azure/docs/getting-started/installation/[Azure Keyvault Provider].
+// TODO: Is there only a portion of these docs that they should follow? And any additional steps they need to take for OCP?
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Create a secret provider class to define your secrets store provider.
+
+.. Create a YAML file that defines the `SecretProviderClass` object:
++
+.Example `secret-provider-class.yaml`
+[source,yaml]
+----
+apiVersion: secrets-store.csi.x-k8s.io/v1
```

Comment:
@ropatil010 @knelasevero

What config do we want to show for the secret provider class for azure? I took most of this from the azure repo. But I didn't see any details on this in the test cases, so if we have something we're already testing with please let me know and we can use it here.

Reply:
Adding tested secretproviderclass.yaml file for reference 

apiVersion: secrets-store.csi.x-k8s.io/v1
kind: SecretProviderClass
metadata:
  name: secretproviderclassname    <1>
  namespace: testropatil // Any randome name 
spec:
  provider: azure
  parameters:
    usePodIdentity: "false"
    useVMManagedIdentity: "false"
    userAssignedIdentityID: ""
    keyvaultName: "kvname"
    objects: |
      array:
        - |
          objectName: secret1 // Any random names 
          objectType: secret
        - |
          objectName: secret2 // Any random names 
          objectType: secret
        - |
          objectName: secret3 // Any random names 
          objectType: secret
    tenantId: "tid"   

// If working with multiple keys we can add lilke:
      array:
        - |
          objectName: key1  // Any random names 
          objectType: key                                          
        - |
          objectName: key2 // Any random names 
          objectType: key  
        - |
          objectName: key3 // Any random names 
          objectType: key
 
 // Multiple certificates         
       array:
        - |
          objectName: cert1 // Any random names 
          objectType: cert                           
        - |
          objectName: cert2 // Any random names 
          objectType: cert                      
        - |
          objectName: cert3 // Any random names 
          objectType: cert

Reply:
Above pasted is not with correct yaml format. Adding [reference link](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/examples/keyvault-keys/v1alpha1_secretproviderclass_keys.yaml) for creating secretproviderclass with multiple keys

Reply:
Thanks, I added in a few fields that you had that I didn't, and added a namespace. I kept the object the same just because I think it coordinated with the section have on enabling sync. But let me know if you think I should show multiple things here in this example.